garak/generators/base.py

```diff
@@ -162,13 +163,27 @@ def generate(
                 )
                 multi_generator_bar.set_description(self.fullname[:55])
 
-                with Pool(_config.system.parallel_requests) as pool:
-                    for result in pool.imap_unordered(
-                        self._call_model, [prompt] * generations_this_call
-                    ):
-                        self._verify_model_result(result)
-                        outputs.append(result[0])
-                        multi_generator_bar.update(1)
+                pool_size = min(
+                    generations_this_call,
+                    _config.system.parallel_requests,
+                    _config.system.max_workers,
+                )
```

Comment:
Direct access to config here suggest we should have a helper that owns process or thread pools that references `_config.system`.

Reply:
i think this is now resolved